modules/dnn/src/layers/batch_norm_layer.cpp

```diff
@@ -178,11 +181,11 @@ class BatchNormLayerImpl CV_FINAL : public BatchNormLayer
         }
 
         UMat &inpBlob = inputs[0];
-        CV_Assert(inpBlob.dims == 2 || inpBlob.dims == 4);
         int groups = inpBlob.size[0];
         int channels = inpBlob.size[1];
-        int rows = inpBlob.dims > 2 ? inpBlob.size[2] : 1;
-        int cols = inpBlob.dims > 2 ? inpBlob.size[3] : 1;
+        int depth = inpBlob.dims == 5 ? inpBlob.size[2] : 1;
+        int rows = inpBlob.dims > 2 ? inpBlob.size[inpBlob.dims - 2] : 1;
+        int cols = inpBlob.dims > 2 ? inpBlob.size[inpBlob.dims - 1] : 1;
```

Comment:
Why not just compute a total number of elements in spatial dimensions?

```cpp
int planeSize = total(inpBlob.size, 2);
Mat inpBlobPlane(1, planeSize, CV_32F, inpBlob.ptr<float>(num, n));
Mat outBlobPlane(1, planeSize, CV_32F, outBlob.ptr<float>(num, n));
...

```

Reply:
done